proxy/src/source.rs

```diff
@@ -91,11 +92,11 @@ impl From<radicle_registry_client::Project> for Project {
 pub trait Source {
```

Comment:
I would actually do one of two things here:
1. Parameterise the trait with a generic `Error`, i.e. `Source<Error>`
2. Add an associated `Error` type.

The reason being that traits should be general to the implementation, and `RegistryError` is specific to the final `registry` implementation. The benefit is that you may want to a test a "pure" implementation where you have guaranteed data and you can set the `Error` to just be `()`.